kernel/src/mm/guestmem.rs

```diff
@@ -39,9 +39,21 @@ pub fn read_u8(v: VirtAddr) -> Result<u8, SvsmError> {
     }
 }
 
+/// Writes 1 byte at a virtual address.
+///
+/// # Safety
+///
+/// The caller should verify that the write address (`v`) is valid for write:
+///  - `v` should belong to a mapped memory region
+///  - `v` should only belong to the issuer's memory space (a guest should not be
+///    able to write in the SVSM kernel's memory, or a different guest's
+///    memory).
+///
+/// This function returns an error if it tries to write to an unmapped address,
+/// and that the #PF or #GP handler returns an error.
```

Comment:
This sentence does not make much sense I think, and also talks about exceptions, which are an implementation detail the caller does not need to know about. I'd simply say:

```suggestion
/// # Returns
///
/// Returns an error if the specified address is not mapped or is not mapped
/// with the appropriate write permissions.
```

Reply:
Applied the suggestion.